### ISSUE TITLE
Payment slip embedding

### ISSUE THREAD

**Issue by marten66** [sprain/php-swiss-qr-bill#94]
I would like to include the payment slip in an invoice that I have generated via fpdf. On the second page with a text on top. I solved it in such a way: I converted the PDF output slip with ImageMagick into an image and then embed it in my fpdf invoice. But the resolution is poor. What possibilities are there to implement this with a good resolution? It runs via a REST Api.

**Comment by Franco2911** [comment]
You can create a new _YourPdfModel_ create a footerSlip function in it with the provided fpdf slip. It works fine for us. Just add a Nextpage before.

**Comment by sprain** [comment]
Yep, what @Franco2911 explained is the way to go with FPDF. Good luck!

**Comment by marten66** [comment]
Unfortunately, this short answer doesn't really help me. As far as I know, it is not possible to import a pdf into FPDF. Can you post me a short sample code.

**Comment by Franco2911** [comment]
It is not so difficult...
if you have a look to the fpdf-example.php in example folder of this library you will understand how it work.

Just change this

```
// 2. Create an FPDF instance (or use an existing one from your project)
$fpdf = new \Fpdf\Fpdf('P', 'mm', 'A4');
$fpdf->AddPage();
```
with your own pdf_model `$pdf = new Your_Namespace_To_Your_Model_Class`

Your model must extends Fpdf.

Good luck!

**Comment by marten66** [comment]
I see. In this case, I have to integrate the library in my application. My problem with this is that my application runs under PHP 5.6 and cannot be updated at the moment, and this library does not run under 5.6. That is why I have outsourced the library and the payment slip is then integrated into the application using an API. I have now solved it as follows. I convert the PDF payment slip using Gohstscript, this allows a higher resolution than Imagick. Then I cut the generated image with Imagick. The generated slip image can then be integrated by the application using the API.

// convert pdf to image with gohstscript
$shellCommand = "gs -dSAFER -dBATCH -dNOPAUSE -sDEVICE=png16m -r300 -sOutputFile=$imageOutput $pfdInput";
exec($shellCommand);

// crop image
$imagick = new \Imagick(realpath($imageOutput));
$imagick->cropImage(2480, 1000, 0, 2300);
$imagick->writeImages($imageOutput, false);
